src/main/java/net/goldiriath/plugin/game/mobspawn/MobSpawnProfile.java

```diff
@@ -150,30 +148,81 @@ public NPC spawn(Location location) {
         if (entityTypeName == null) {
             type = null;
         } else {
-            type = EntityType.fromName(entityTypeName);
+            try {
+                type = EntityType.valueOf(entityTypeName);
+            } catch (IllegalArgumentException ex) {
+                type = null;
+                logger.warning("Could not load mobspawn profile '" + id + "'. Invalid entity type: " + entityTypeName);
+            }
+        }
+
+        // Stats
+        level = config.getInt("stats.level", -1);
+        if (level == -1) {
+            logger.warning("Could not load mobspawn profile '" + id + "'. Level field not present.");
+        }
+        damage = config.getInt("stats.damage", -1);
+        health = config.getInt("stats.health", -1);
+        if (health == -1) {
+            logger.warning("Could not load mobspawn profile '" + id + "'. Health field not present.");
+        }
+
+        // Loot
+        if (!config.contains("loot")) {
+            lootProfile = null;
+            lootTier = null;
+        } else {
+            String profileId = config.getString("loot.profile", null);
+            lootProfile = plugin.lm.getProfile(profileId);
+            if (lootProfile == null) {
+                logger.warning("Could not load mobspawn profile '" + id + "'. Invalid loot table id: " + profileId);
+            }
+            String tierName = config.getString("loot.tier", null);
+            try {
+                lootTier = MobTier.valueOf(tierName);
+            } catch (Exception ex) {
+                lootTier = null;
+                logger.warning("Could not load mobspawn profile '" + id + "'. Invalid loot table id: " + tierName);
+
+            }
         }
 
-        // Meta
-        customName = config.getString("name", null);
-        timeThreshold = config.getInt("spawn_threshold", -1);
-        level = config.getInt("level", 1);
-        tier = MobTier.valueOf(config.getString("tier", MobTier.NORMAL.name()));
-        loot = manager.getPlugin().lm.getProfileMap().get(config.getString("loot_profile", "default"));
+        // Items
+        if (!config.contains("items")) {
+            hand = null;
+            helmet = null;
+            chestplate = null;
+            leggings = null;
+            boots = null;
+        } else {
+            hand = Util.parseItem(config.getString("items.hand", null));
+            helmet = Util.parseItem(config.getString("items.helmet", null));
+            chestplate = Util.parseItem(config.getString("items.chestplate", null));
+            leggings = Util.parseItem(config.getString("items.leggings", null));
+            boots = Util.parseItem(config.getString("items.boots", null));
+        }
+
+        if (damage == -1 && hand == null) {
+            logger.warning("Could not load mobspawn profile '" + id + "'. Damage field not present and item not present. Must specify either to calculate attack damage.");
+        }
```

Comment:
here again, there is a warning given, but the method still continues, is that alright?